### PR TITLE
add ERNIE-CTM configuration class and move ERNIE-CTM modeling and rel…

### PR DESCRIPTION
**Pull request by qhpeklh5959**
…ated tasks onto configuration.

### PR types

New Features

### PR changes

APIs

### Description

Add ERNIE-CTM configuration class and move modeling and related tasks onto configuration-based initializations.

### CODE REVIEW

File: paddlenlp/transformers/ernie_ctm/configuration.py [hunk 1]
```diff
@@ -0,0 +1,140 @@
...
+        "type_vocab_size": 2,
+        "initializer_range": 0.02,
+        "pad_token_id": 0,
+        "use_content_summary": True,
+        "content_summary_index": 1,
+        "cls_num": 2,
+        "num_prompt_placeholders": 5,
+        "prompt_vocab_ids": None
+    },
+}
```
Comment (wj-Mcat) [inline paddlenlp/transformers/ernie_ctm/configuration.py]: 虽然还是配置是一样的，可是这里的key还是需要包含: "wordtag"和"nptag"的，不然后面在from_pretrained的时候会出错。所以在这里我建议用以下写法：

```python
ERNIE_CTM_CONFIG = {
    "vocab_size": 23000,
    ......
}
ERNIE_CTM_PRETRAINED_INIT_CONFIGURATION = {
    "ernie-ctm": ERNIE_CTM_CONFIG,
    "wordtag": ERNIE_CTM_CONFIG,
    "nptag": ERNIE_CTM_CONFIG
}
```

因为这个属性在configuration.py里面需要配置，在modeling.py里面也是需要配的，而且都需要三个字段，所以综合考虑，用这种方式比较简洁。您觉得呢？

此外，我这里有一个小疑问：因为都是下面两个都是基于 ernie-ctm 模型，带上一个前缀呢？这样也避免以后和其他模型有冲突，比如模型名称调整成："ernie-ctm-wordtag", "ernie-ctm-nptag"等。

File: paddlenlp/transformers/ernie_ctm/configuration.py [hunk 2]
```diff
@@ -0,0 +1,140 @@
...
+            If `use_content_summary` is set, content summary token position is defined by this argument.
+        cls_num (`int`, *optional*, defaults to 2):
+            Number of [CLS] token in model.
+        num_prompt_placeholders (`int`, *optional*, defaults to 5):
+            Number of maximum length of prompt answer.
+        prompt_vocab_ids (`dict`, *optional*, defaults to None):
+            Prompt vocabulary of decode procedure.
+    """
+    model_type = "ernie-ctm"
+    pretrained_init_configuration = ERNIE_CTM_PRETRAINED_INIT_CONFIGURATION
```
Comment (wj-Mcat) [inline paddlenlp/transformers/ernie_ctm/configuration.py:103]: 如果用原本的写法，这里字典里面的 key 将会只有一个，会出现问题。

File: paddlenlp/transformers/ernie_ctm/modeling.py [hunk 3]
```diff
@@ -111,70 +145,18 @@ class ErnieCtmPretrainedModel(PretrainedModel):
...
-            "type_vocab_size": 2,
-            "initializer_range": 0.02,
-            "pad_token_id": 0,
-            "use_content_summary": True,
-            "content_summary_index": 1,
-            "cls_num": 2,
-        },
+        "ernie-ctm": config,
+        "wordtag": config,
+        "nptag": config,
```
Comment (wj-Mcat) [inline paddlenlp/transformers/ernie_ctm/modeling.py]: 这部分的重复性配置可以在configuration.py文件里面就做好。

File: paddlenlp/transformers/ernie_ctm/modeling.py [hunk 4]
```diff
@@ -254,55 +237,39 @@ class ErnieCtmModel(ErnieCtmPretrainedModel):
...
-
-        self.encoder = nn.TransformerEncoder(encoder_layer, num_hidden_layers)
-        self.pooler = ErnieCtmPooler(hidden_size)
-
-        self.use_content_summary = use_content_summary
-        self.content_summary_index = content_summary_index
-        if use_content_summary is True:
-            self.feature_fuse = nn.Linear(hidden_size * 2, intermediate_size)
-            self.feature_output = nn.Linear(intermediate_size, hidden_size)
+    def __init__(self, config: ErnieCtmConfig, *model_args, **kwargs):
```
Comment (wj-Mcat) [inline paddlenlp/transformers/ernie_ctm/modeling.py]: 这里使用*model_args和**kwargs是为了防止其他没必要的参数传递进来，可是这种情况已经在基类中处理好了的，如果还有问题，可以反馈，因为这是一个bug。

File: paddlenlp/transformers/ernie_ctm/modeling.py [hunk 5]
```diff
@@ -471,7 +474,10 @@ def forward(self,
                 lengths=None,
                 position_ids=None,
                 attention_mask=None,
-                tag_labels=None):
+                tag_labels=None,
```
Comment (wj-Mcat) [inline paddlenlp/transformers/ernie_ctm/modeling.py:482]: 在框架这边，都统一参数为：labels，您看这里的tag_labels改成 labels 是否更合适些呢？

**Review by wj-Mcat — CHANGES_REQUESTED** [review]

**Review by wj-Mcat — DISMISSED** [review]

**Review by wj-Mcat — APPROVED** [review]
